features/dd-sdk-android-trace/src/test/kotlin/com/datadog/opentracing/PendingTraceTest.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.opentracing
+
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.trace.api.sampling.PrioritySampling
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+import java.math.BigInteger
+import java.util.ConcurrentModificationException
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(value = Configurator::class)
+class PendingTraceTest {
+
+    @Mock
+    lateinit var mockTracer: DDTracer
+
+    lateinit var fakeTraceId: BigInteger
+
+    @Test
+    fun `PendingTrace is thread safe`(forge: Forge) {
+        whenever(mockTracer.partialFlushMinSpans) doReturn 1
+        fakeTraceId = BigInteger.valueOf(forge.aLong())
+        val pendingTrace = PendingTrace(mockTracer, fakeTraceId)
+        val rootSpan = forge.fakeSpan(pendingTrace, mockTracer, fakeTraceId, BigInteger.ZERO)
+        val countDownLatch = CountDownLatch(2)
+
+        val runnables = Array(4) {
+            StressTestRunnable(pendingTrace, mockTracer, rootSpan, forge, countDownLatch)
+        }
+        runnables.forEach { Thread(it).start() }
+
+        countDownLatch.await(10, TimeUnit.SECONDS)
```

Comment:
this may make test flaky, shouldn't we just use simple `await` knowing that all runnables will complete anyway?

Reply:
> all runnables will complete anyway?
Actually that's not true: if an exception happens in the Runnable other than `ConcurrentModificationException`, then the thread will crash without crashing the test run, and the `countDownLatch` won't be decreased. 
I timed the tests as taking 1.5 seconds, so 10 seconds gives a lot of margin to prevent flakiness.